python/pyspark/sql/context.py

```diff
@@ -202,6 +202,10 @@ def registerFunction(self, name, f, returnType=StringType()):
         """
         self.sparkSession.catalog.registerFunction(name, f, returnType)
 
+    def registerJavaFunction(self, name, javaClassName, returnType):
```

Comment:
It would be good to have tests that call this from the Python side, also it probably needs a docstring and since annotation as well (see registerFunction since you take similar params).
It might also make sense to have the same default return type of StringType as the current registerFunction but that is more a matter of taste.